sql/src/main/java/io/crate/analyze/validator/LiteralVisitor.java

```diff
@@ -0,0 +1,31 @@
+package io.crate.analyze.validator;
+
+import io.crate.analyze.symbol.Function;
+import io.crate.analyze.symbol.Literal;
+import io.crate.analyze.symbol.Symbol;
+import io.crate.analyze.symbol.SymbolVisitor;
+
+/**
+ * Returns true if all symbols in an expression are literals and false otherwise.
+ */
+public class LiteralVisitor extends SymbolVisitor<Void, Boolean> {
```

Comment:
I don't like the name LiteralVisitor now that it's public class but also don't have a good suggestion :-(